pulpcore/cli/core/content_guard.py

```diff
@@ -0,0 +1,105 @@
+import gettext
+from typing import List, Optional
+
+import click
+
+from pulpcore.cli.common.context import (
+    PluginRequirement,
+    PulpContext,
+    pass_entity_context,
+    pass_pulp_context,
+)
+from pulpcore.cli.common.generic import (
+    create_command,
+    destroy_command,
+    href_option,
+    list_command,
+    load_json_callback,
+    name_option,
+    show_command,
+    update_command,
+)
+from pulpcore.cli.core.context import PulpContentGuardContext, PulpRbacContentGuardContext
+
+_ = gettext.gettext
+
+
+@click.group()
+@pass_pulp_context
+@click.pass_context
+def content_guard(ctx: click.Context, pulp_ctx: PulpContext) -> None:
+    ctx.obj = PulpContentGuardContext(pulp_ctx)
+
+
+create_options = [click.option("--name", required=True), click.option("--description")]
+filter_options = [click.option("--name")]
+lookup_options = [name_option, href_option]
+
+content_guard.add_command(list_command(decorators=filter_options))
+
+
+@content_guard.group()
+@pass_pulp_context
+@click.pass_context
+def rbac(ctx: click.Context, pulp_ctx: PulpContext) -> None:
+    pulp_ctx.needs_plugin(PluginRequirement("core", "3.15.0.dev"))
+    ctx.obj = PulpRbacContentGuardContext(pulp_ctx)
+
+
+rbac.add_command(list_command(decorators=filter_options))
+rbac.add_command(create_command(decorators=create_options))
+rbac.add_command(show_command(decorators=lookup_options))
+rbac.add_command(update_command(decorators=lookup_options))
+rbac.add_command(destroy_command(decorators=lookup_options))
+
+
+@rbac.command()
+@name_option
+@href_option
+@click.option(
+    "--groups",
+    help=_("List of groups to assign download permission to"),
+    callback=load_json_callback,
+)
+@click.option(
+    "--users",
+    help=_("List of users to assign download permission to"),
+    callback=load_json_callback,
+)
+@pass_entity_context
+@pass_pulp_context
+def assign(
+    pulp_ctx: PulpContext,
+    guard_ctx: PulpRbacContentGuardContext,
+    users: Optional[List[str]],
+    groups: Optional[List[str]],
+) -> None:
+    href = guard_ctx.entity["pulp_href"]
+    result = guard_ctx.assign(href=href, users=users, groups=groups)
+    pulp_ctx.output_result(result)
+
+
+@rbac.command()
+@name_option
+@href_option
+@click.option(
+    "--groups",
+    help=_("List of groups to remove download permission from"),
+    callback=load_json_callback,
+)
```

Comment:
Regarding my other comment, i think this would be the change:

```suggestion
@click.option(
    "--group",
    "groups",
    help=_("Groups to remove download permission from. Can be specified multiple times."),
    multiple=True,
)
```